toolchain/lower/handle.cpp

```diff
@@ -339,6 +339,39 @@ static auto HandleBuiltinCall(FunctionContext& context, SemIR::InstId inst_id,
                            context.GetValue(arg_ids[1]), "cmp"));
       return;
     }
+    case SemIR::BuiltinFunctionKind::FloatNegate: {
+      // Lower `-x` as `0 - x`
+      auto* operand = context.GetValue(arg_ids[0]);
+      context.SetLocal(inst_id,
+                       context.builder().CreateFSub(
+                           llvm::ConstantInt::getNullValue(operand->getType()),
+                           operand, "fneg"));
+      return;
```

Comment:
Use `CreateFNeg` here instead of `CreateFSub`. Unlike for integers, `0.0 - x` isn't equivalent to `- x`, specifically when `x` is `0.0` -- `-0.0` should give a negative zero, but `0.0 - 0.0` is a positive zero.

Reply:
Done! Thanks for the context!